fusilli-plugin/include/graph_import.h

```diff
@@ -0,0 +1,247 @@
+#define FUSILLI_PLUGIN_SRC_GRAPH_IMPORT_H
+
+#include <fusilli.h>
+#include <hipdnn_sdk/data_objects/data_types_generated.h>
+#include <hipdnn_sdk/data_objects/tensor_attributes_generated.h>
+#include <hipdnn_sdk/plugin/PluginApiDataTypes.h>
+#include <hipdnn_sdk/plugin/flatbuffer_utilities/GraphWrapper.hpp>
+
+#include <format>
+#include <memory>
+#include <unordered_map>
+
+#include "fusilli/support/logging.h"
+#include "hipdnn_engine_plugin_execution_context.h"
+
+// Convert from hipDNN DataType to fusilli DataType
+inline fusilli::ErrorOr<fusilli::DataType>
+convertHipDnnToFusilli(hipdnn_sdk::data_objects::DataType hipdnnType) {
+  switch (hipdnnType) {
+  case hipdnn_sdk::data_objects::DataType::HALF:
+    return ok(fusilli::DataType::Half);
+  case hipdnn_sdk::data_objects::DataType::BFLOAT16:
+    return ok(fusilli::DataType::BFloat16);
+  case hipdnn_sdk::data_objects::DataType::FLOAT:
+    return ok(fusilli::DataType::Float);
+  case hipdnn_sdk::data_objects::DataType::DOUBLE:
+    return ok(fusilli::DataType::Double);
+  case hipdnn_sdk::data_objects::DataType::UINT8:
+    return ok(fusilli::DataType::Uint8);
+  case hipdnn_sdk::data_objects::DataType::INT32:
+    return ok(fusilli::DataType::Int32);
+  case hipdnn_sdk::data_objects::DataType::UNSET:
+    return ok(fusilli::DataType::NotSet);
+  default:
+    return error(fusilli::ErrorCode::RuntimeFailure,
+                 "Unknown type in hipdnn -> fusilli graph translation.");
+  }
+}
+
+// Graph import is done through importGraph function, this class exists for
+// organization and is used by importGraph.
+//
+// Graph import is designed around individual Node import functions (such as
+// importConvFPropAttr) which convert a given node type, and track input and
+// output tensors in shared state (via importNodeInput and importNodeOutput
+// functions). Graph nodes are processed in topological order to ensure that
+// outputs of producer nodes are tracked and available for consuming nodes.
+//
+// NOTE: topological ordering not yet implemented. As we're only handling
+// single-node graphs currently, that's not a problem.
+class GraphImport {
+private:
+  friend fusilli::ErrorOr<HipdnnEnginePluginExecutionContext>
+  importGraph(const hipdnnPluginConstData_t *opGraph);
+
+  // The imported graph.
+  fusilli::Graph fusilliGraph;
+
+  // Maps hipDNN tensor UIDs to fusilli::TensorAttrs for graph boundary tensors
+  // (inputs and outputs). Used by hipdnnEnginePluginExecuteOpGraph to match
+  // incoming device buffers (identified by UID) to their corresponding
+  // fusilli::TensorAttr.
+  //
+  // All tensors in this map should be non-virtual boundary tensors---a virtual
+  // tensor is an internal intermediate---internal tensors will be tracked
+  // elsewhere. Currently we only support single node graphs, so there are no
+  // virtual tensors to track.
+  std::unordered_map<int64_t, std::shared_ptr<fusilli::TensorAttr>>
+      uidToIOTensor;
```

Comment:
This is maybe a question for the hipDNN folks (@BrianHarrisonAMD) - How prevalent is the UID based variantPack compared to the name based variantPack? I see both being exposed in cudnn frontend. In fusilli, we're only supporting name based. I guess the plugin here is basically trying to map from UID -> name (which is not too bad) but if this is something we can do without, we should clean it up.

Reply:
> I am not 100% sure how this will work on your side with the intermediates and the TensorAttrs for Fusilli, but for the hipDNN schema, virtual tensors will also have Ids, but just won't have a variant pack device buffer.

I think we're aligned well here, any intermediates will be tracked for graph translation, but when it comes to execution we will only care about the id <> `fusilli::TensorAttr` mapping so we can properly map input and output buffers.